Nodejs/Product/TestAdapterImpl/TestContainerDiscoverer.cs

```diff
@@ -304,8 +320,8 @@ private bool ShouldDiscover(string pathToItem)
             }
 
             //Setting/updating "TestFramework" property on a file item will cause metedata change in the project file,
-            //so we need to re-discover when file change happens. Since we support all project files, this is a safe check
-            if (pathToItem.EndsWith("proj", StringComparison.OrdinalIgnoreCase))
+            //so we need to re-discover when file change happens.
+            if (TypeScriptHelpers.IsProjectFile(pathToItem))
```

Comment:
I noticed. But since we're registering the test for only three project types (csproj, njsproj and vbproj) I think is better to specify them. Otherwise we might have an issue in the future.

Reply:
Excellent, that makes perfect sense.  Can you please add comments to `IsSupportedTestProjectFile` and the registration attributes indicating that they need to stay in sync?

Reply:
Added.